app/src/main/java/com/sun/moviedb/data/repository/MovieByCategoryRepository.kt

```diff
@@ -0,0 +1,47 @@
+package com.sun.moviedb.data.repository
+
+import androidx.lifecycle.LiveData
+import com.sun.moviedb.base.BaseResponse
+import com.sun.moviedb.base.ContextProviders
+import com.sun.moviedb.base.NetworkBoundResource
+import com.sun.moviedb.data.DatabaseManager
+import com.sun.moviedb.data.dto.CategoryDto
+import com.sun.moviedb.data.dto.MovieDto
+import com.sun.moviedb.data.entity.Movie
+import com.sun.moviedb.data.remote.RemoteDataSource
+import com.sun.moviedb.utils.CategoryQuery
+
+/**
+ * Created by nguyenxuanhoi on 2019-08-20.
+ * @author nguyen.xuan.hoi@sun-asterisk.com
+ */
+class MovieByCategoryRepository(val coroutines: ContextProviders, private val remoteDataSource: RemoteDataSource) :
```

Comment:
```suggestion
class MovieByCategoryRepository(
    val coroutines: ContextProviders,
    private val remoteDataSource: RemoteDataSource
) : MovieByCategoryContract {
```
cú pháp kotlin viết ntn em nhé
k như java
tham khảo convention 
https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting